src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs

```diff
@@ -0,0 +1,170 @@
+            SyntaxNode node = root.FindNode(context.Span);
+            SemanticModel model = await document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, context.CancellationToken).ConfigureAwait(false);
+            SyntaxGenerator generator = editor.Generator;
+
+            string title = MicrosoftNetCoreAnalyzersResources.AvoidConstArraysCodeFixTitle;
+            context.RegisterCodeFix(CodeAction.Create(
+                    title,
+                    async c => await ExtractConstArrayAsync(node, model, editor, generator, context.Diagnostics.First().Properties, c).ConfigureAwait(false),
+                    equivalenceKey: title),
+                context.Diagnostics);
+        }
+
+        private static async Task<Document> ExtractConstArrayAsync(SyntaxNode node, SemanticModel model, DocumentEditor editor,
+            SyntaxGenerator generator, ImmutableDictionary<string, string> properties, CancellationToken cancellationToken)
+        {
+            IArrayCreationOperation arrayArgument = GetArrayCreationOperation(node, model, cancellationToken, out bool isInvoked);
+            IEnumerable<ISymbol> members = model.GetEnclosingSymbol(node.SpanStart, cancellationToken).ContainingType.GetMembers();
+
+            // Get a valid member name for the extracted constant
+            string newMemberName = GetExtractedMemberName(
+                members.Where(x => x is IFieldSymbol).Select(x => x.Name),
+                properties["paramName"] ?? GetMemberNameFromType(arrayArgument));
+
+            // Get method containing the symbol that is being diagnosed.
+            IOperation? methodContext = arrayArgument.GetAncestor<IMethodBodyOperation>(OperationKind.MethodBody);
+            methodContext ??= arrayArgument.GetAncestor<IBlockOperation>(OperationKind.Block); // VB methods have a different structure than CS methods
+
+            // Create the new member
+            SyntaxNode newMember = generator.FieldDeclaration(
+                newMemberName,
+                generator.TypeExpression(arrayArgument.Type),
+                GetAccessibility(methodContext is null ? null : model.GetEnclosingSymbol(methodContext.Syntax.SpanStart, cancellationToken)),
+                DeclarationModifiers.Static | DeclarationModifiers.ReadOnly,
+                arrayArgument.Syntax.WithoutTrailingTrivia() // don't include extra trivia before the end of the declaration
+            );
+
+            // Add any additional formatting
+            if (methodContext is not null)
+            {
+                newMember = newMember.FormatForExtraction(methodContext.Syntax);
+            }
+
+            // Insert the new extracted member before the first member
+            SyntaxNode firstMemberNode = await members.First().DeclaringSyntaxReferences.First()
```

Comment:
I'm usually trying to play defensive and try to favor `FirstOrDefault` over `First` just in case we have an invalid tree.